### ISSUE TITLE
UselessParenthesesSniff

### ISSUE THREAD

**Issue by kukulich** [slevomat/coding-standard#374]
https://github.com/doctrine/doctrine2/pull/7297#discussion_r200220249

**Comment by kukulich** [comment]
Implemented in https://github.com/slevomat/coding-standard/commit/f7feec44cdaf9d9b290217afdd03d47978740e43

I'm open for more ideas what should be reported as useless parentheses.

**Comment by carusogabriel** [comment]
@kukulich Looks like this wasn't catch https://github.com/doctrine/coding-standard/pull/70/files#diff-ed68ee80435e5b99183806044803ac8bR21

**Comment by kukulich** [comment]
@carusogabriel Please try again :)

**Comment by carusogabriel** [comment]
@kukulich Thank you, it works. I'll test in a large codebase during the week and give you feedback in more cases we can use it, or it's fine as it's. I've found a bug:

```php
return new class ($number) { // The Sniff reports and removes this parentheses :(
}
```

**Comment by kukulich** [comment]
@carusogabriel Fixed :)

**Comment by kukulich** [comment]
@carusogabriel Done

**Comment by Levivb** [comment]
I've found another false positive:
```php
$rowData[] = $row['function']($item);
```
It 'fixes' it to:
```php
$rowData[] = $row['function']$item;
```
which of course is invalid php

**Comment by kukulich** [comment]
Fixed in https://github.com/slevomat/coding-standard/commit/fbdda3e0d7b6a12c4964778a3443c3883d9503a3
